model/internal/pdata/common.go

```diff
@@ -186,27 +186,27 @@ func (v Value) BoolVal() bool {
 }
 
 // MapVal returns the map value associated with this Value.
-// If the Type() is not ValueTypeMap then returns an empty map. Note that modifying
-// such empty map has no effect on this Value.
+// If the Type() is not ValueTypeMap then returns an invalid map. Note that using
+// such map can cause panic.
 //
 // Calling this function on zero-initialized Value will cause a panic.
 func (v Value) MapVal() AttributeMap {
 	kvlist := v.orig.GetKvlistValue()
 	if kvlist == nil {
-		return NewAttributeMap()
+		return AttributeMap{}
 	}
 	return newAttributeMap(&kvlist.Values)
 }
```

Comment:
```golang
func (v Value) MapVal() AttributeMap {
	kvlist, ok := v.orig.GetValue().(*otlpcommon.AnyValue_KvlistValue)
	if !ok {
		return AttributeMap{}
	}
	return newAttributeMap(&kvlist.KvlistValue.Values)
}
```

Reply:
Hm... `v.orig.Get<other>Value()` essentially implement the same instructions. So I'll just update these two

Reply:
done